parser.js

```diff
@@ -176,7 +179,28 @@ function generateAST(code, options, additionalParsingContext) {
     const ast = program.getSourceFile(FILENAME);
 
     extra.code = code;
-    return convert(ast, extra);
+    return {
+        ast: convert(ast, extra),
+        program
+    };
+}
+
+/**
+ * Generate the `parserServices` object for a parse result
+ * @param   {Result} result The return value of `generateAST`
+ * @returns {Object}        The `parserServices` object
+ */
+function getServices(result) {
+    const typeChecker = result.program.getTypeChecker();
+    return {
+        program: result.program,
+        getTSNode: node => nodeUtils.map.get(node),
+        getESNode: tsNode => nodeUtils.reverseMap.get(tsNode),
```

Comment:
Again, `ESTreeNode` is the name that has been used.

I'm a little unsure about the use-case for `getESNode` and `reverseMap` as they are currently called. I feel which should err on adding as little as possible to this API surface, particularly in this initial PR, we can always add more later